homeassistant/components/auth/login_flow.py

```diff
@@ -101,11 +102,40 @@ async def async_setup(
     hass: HomeAssistant, store_result: Callable[[str, Credentials], str]
 ) -> None:
     """Component to allow users to login."""
+    hass.http.register_view(WellKnownOAuthInfoView)
     hass.http.register_view(AuthProvidersView)
     hass.http.register_view(LoginFlowIndexView(hass.auth.login_flow, store_result))
     hass.http.register_view(LoginFlowResourceView(hass.auth.login_flow, store_result))
 
 
+class WellKnownOAuthInfoView(HomeAssistantView):
+    """View to host the OAuth2 information."""
+
+    requires_auth = False
+    url = "/.well-known/oauth-authorization-server"
+    name = "well-known/oauth-authorization-server"
+
+    async def get(self, request: web.Request) -> web.Response:
+        """Return the well known OAuth2 authorization info."""
+        hass: HomeAssistant = request.app["hass"]
+
+        try:
+            base_url = network.get_url(hass, require_current_request=True)
+        except network.NoURLAvailableError:
+            base_url = network.get_url(hass, prefer_external=True)
```

Comment:
Could this potentially leak internal network details?

Reply:
I guess if you are to access it over it's external IP address, it could respond with either internal or external configured URL. So yeah, that's not good. Removing the endpoint.